src/Mapping/Parameter/ITypeMapper.php

```diff
@@ -2,13 +2,15 @@
 
 namespace Apitte\Core\Mapping\Parameter;
 
+use Apitte\Core\Schema\EndpointParameter;
+
 interface ITypeMapper
 {
 
 	/**
 	 * @param mixed $value
 	 * @return mixed
 	 */
-	public function normalize($value);
+	public function normalize($value, EndpointParameter $parameter);
```

Comment:
Is EndpointParameter just for exception messsage here?

Reply:
I don't like it. I would prefer try/catch with recreate it.  Mapper would raise special exception and the layer above will transfer it to human readable. WDYT?

Reply:
Look at all error messages. They also contains some hints for specific type. How would you do it in the layer above? I know it would be possible with $parameter->getIn and list of messages but how would be someone able to use datetime in not ISO 8601 format if type conversion is in one layer and message is in another?